drivers/pwm/pwm_stm32.c

```diff
@@ -753,7 +759,10 @@ static void pwm_stm32_irq_config_func_##index(const struct device *dev)		\
 	}
 
 #define PWM_DEVICE_INIT(index)                                                 \
-	static struct pwm_stm32_data pwm_stm32_data_##index;                   \
+	static struct pwm_stm32_data pwm_stm32_data_##index = {                \
+		.reset = RESET_DT_SPEC_GET(DT_INST_PARENT(index)),	       \
+	};								       \
```

Comment:
Like in the `counter_ll_stm32_timer.c`, it could written adding a macro 
`#define PWM(idx)              DT_INST_PARENT(idx)`